File: apps/ruby-sdk/lib/firecrawl/models/pdf_parser.rb
# frozen_string_literal: true

module Firecrawl
  module Models
    # Configuration for parsing PDF documents.
    class PdfParser
      attr_reader :mode, :max_pages, :page_markdown

      def initialize(mode: nil, max_pages: nil, page_markdown: nil)
        @mode = mode
        @max_pages = max_pages
        @page_markdown = page_markdown
      end

      def to_h
        {
          "type" => "pdf",
          "mode" => mode,
          "maxPages" => max_pages,
          "pageMarkdown" => page_markdown,
        }.compact
      end
    end
  end
end

File: apps/dot-net-sdk/Firecrawl/Models/PdfParser.cs
using System.Text.Json.Serialization;

namespace Firecrawl.Models;

/// <summary>
/// Configuration for parsing PDF documents.
/// </summary>
public class PdfParser
{
    [JsonPropertyName("type")]
    public string Type { get; } = "pdf";

    [JsonPropertyName("mode")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? Mode { get; set; }

    [JsonPropertyName("maxPages")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public int? MaxPages { get; set; }

    [JsonPropertyName("pageMarkdown")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public bool? PageMarkdown { get; set; }
}
